### PR TITLE
Remove redundant assignments (dead stores)

### PR DESCRIPTION
**Pull request by practicalswift**
Remove redundant assignments (dead stores).

### CODE REVIEW

**Comment by Empact** [comment]
utACK 3e0fd18

**Comment by promag** [comment]
utACK 3e0fd18.

**Comment by paveljanik** [comment]
utACK 3e0fd18

File: src/bench/crypto_hash.cpp [hunk 1]
```diff
@@ -80,18 +80,16 @@ static void SipHash_32b(benchmark::State& state)
...
     }
 }
 
 static void FastRandom_1bit(benchmark::State& state)
 {
     FastRandomContext rng(true);
-    uint32_t x = 0;
     while (state.KeepRunning()) {
-        x += rng.randbool();
+        rng.randbool();
```
Comment (MarcoFalke) [inline src/bench/crypto_hash.cpp]: Have you checked if this changes the benchmark result. If yes, how much?

File: src/bench/rollingbloom.cpp [hunk 2]
```diff
@@ -25,7 +24,7 @@ static void RollingBloom(benchmark::State& state)
         data[1] = count >> 16;
         data[2] = count >> 8;
         data[3] = count;
-        match += filter.contains(data);
+        filter.contains(data);
```
Comment (MarcoFalke) [inline src/bench/rollingbloom.cpp]: Same here

File: src/txmempool.cpp [hunk 3]
```diff
@@ -645,19 +645,14 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
...
         for (const CTxIn &txin : tx.vin) {
             // Check that every mempool transaction's inputs refer to available coins, or other mempool tx's.
             indexed_transaction_set::const_iterator it2 = mapTx.find(txin.prevout.hash);
             if (it2 != mapTx.end()) {
                 const CTransaction& tx2 = it2->GetTx();
                 assert(tx2.vout.size() > txin.prevout.n && !tx2.vout[txin.prevout.n].IsNull());
                 fDependsWait = true;
-                if (setParentCheck.insert(it2).second) {
-                    parentSizes += it2->GetTxSize();
-                    parentSigOpCost += it2->GetSigOpCost();
```
Comment (MarcoFalke) [inline src/txmempool.cpp]: Instead of removing them, shouldn't they be asserted against `GetSizeWithAncestors` et al?

**Review by maflcko — COMMENTED** [review]
Thanks for taking a look here, but I am not yet convinced just removing them is the obviously correct thing to do.

**Comment by practicalswift** [comment]
Rebased and cherry picked 39ee3e85d12604e10fcded75bb55c2a547158271 into this PR from #13847 as suggested by @fanquake :-)

File: src/txmempool.cpp (hunk 3 above)
Comment (practicalswift) [reply src/txmempool.cpp]: @MarcoFalke I believe that is done later in the function – see  ce019bf90fe89c1256a89c489795987ef0b8a18f. Am I reading it correctly? :-)

Comment (MarcoFalke) [reply src/txmempool.cpp]: Right. Looks good.

File: src/bench/rollingbloom.cpp (hunk 2 above)
Comment (practicalswift) [reply src/bench/rollingbloom.cpp]:
```
$ ./bench_bitcoin-before -filter=RollingBloom
WARNING: This is a debug build - may result in slower benchmarks.
# Benchmark, evals, iterations, total, min, max, median
RollingBloom, 5, 1500000, 13.066, 1.72693e-06, 1.76851e-06, 1.74017e-06
$ ./bench_bitcoin-after -filter=RollingBloom
WARNING: This is a debug build - may result in slower benchmarks.
# Benchmark, evals, iterations, total, min, max, median
RollingBloom, 5, 1500000, 13.2987, 1.75024e-06, 1.82156e-06, 1.76487e-06
```

@MarcoFalke, looks good as well? :-)

**Comment by MarcoFalke** [comment]
utACK dd777f3e12

**Comment by Empact** [comment]
utACK dd777f3

**Comment by practicalswift** [comment]
@promag @paveljanik Would you mind re-reviewing your previous utACK:s? :-)

**Pull request merged**